lib/ui/painting/image_decoder_no_gl_unittests.cc

```diff
@@ -0,0 +1,212 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/painting/image_decoder_no_gl_unittests.h"
+
+namespace flutter {
+namespace testing {
+
+// Tests are disabled for fuchsia.
+#if defined(OS_FUCHSIA)
+#pragma GCC diagnostic ignored "-Wunreachable-code"
+#endif
+
+namespace {
+
+bool IsPngWithPLTE(const uint8_t* bytes, size_t size) {
+  if (size < 8) {
+    return false;
+  }
+
+  if (memcmp(bytes, "\x89PNG\x0d\x0a\x1a\x0a", 8) != 0) {
```

Comment:
use `constexpr std::string_view`  to represent the PNG format string constants and their lengths

Reply:
done